presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```
@@ -0,0 +1,143 @@
+    private final Equivalence<Object> equivalence = Equivalence.identity();
+
+    public IdentityLinkedHashMap()
+    {
+    }
+
+    public IdentityLinkedHashMap(IdentityLinkedHashMap<K, V> map)
+    {
+        putAll(map);
+    }
+
+    @Override
+    public int size()
+    {
+        return delegate.size();
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return delegate.isEmpty();
+    }
+
+    @Override
+    public boolean containsKey(Object key)
+    {
+        return delegate.containsKey(equivalence.wrap(key));
+    }
+
+    @Override
+    public boolean containsValue(Object value)
+    {
+        return delegate.containsValue(value);
+    }
+
+    @Override
+    public V get(Object key)
+    {
+        return delegate.get(equivalence.wrap(key));
+    }
+
+    @Override
+    public V put(K key, V value)
+    {
+        return delegate.put(equivalence.wrap(key), value);
+    }
+
+    @Override
+    public V remove(Object key)
+    {
+        return delegate.remove(equivalence.wrap(key));
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> map)
+    {
+        map.entrySet().forEach(e -> {
+            delegate.put(equivalence.wrap(e.getKey()), e.getValue());
+        });
+    }
+
+    @Override
+    public void clear()
+    {
+        delegate.clear();
+    }
+
+    @Override
+    public Set<K> keySet()
+    {
+        return new AbstractSet<K>()
+        {
+            @Override
+            public Iterator<K> iterator()
+            {
+                return delegate.keySet().stream().map(Equivalence.Wrapper::get).iterator();
+            }
+
+            @Override
+            public int size()
+            {
+                return delegate.size();
+            }
+        };
+    }
+
+    @Override
+    public Collection<V> values()
+    {
+        return delegate.values();
+    }
+
+    @Override
+    public Set<Entry<K, V>> entrySet()
```

The output of this method doesn't satisfy the required contract:

```
* Returns a {@link Set} view of the keys contained in this map.
* The set is backed by the map, so changes to the map are
* reflected in the set, and vice-versa.
```

It would be acceptable to make all other methods throw an exception, but having them silently succeed and not do the right thing is a recipe for future subtle bugs.